sdlib/d/gc/sizeclass.d

```diff
@@ -77,6 +77,11 @@ bool isLargeSize(size_t size) {
 	return (size > SizeClass.Small) && (size <= MaxAllocationSize);
 }
 
+// Determine whether given size class supports appendability.
+bool isAppendableSizeClass(uint sizeClass) {
+	return sizeClass & 1 || sizeClass > 6;
+}
```

Comment:
It's very easy to write a test that checks if this has the properties you expect.